lib/_http_incoming.js

```diff
@@ -121,7 +121,7 @@ IncomingMessage.prototype._addHeaderLines = function(headers, n) {
 // and drop the second. Extended header fields (those beginning with 'x-') are
 // always joined.
 IncomingMessage.prototype._addHeaderLine = function(field, value, dest) {
-  field = field.toLowerCase();
+  field = field.toLowerCase().trim();
```

Comment:
The parser state model is fairly complicated. I get a bit hesitant when it
comes to messing with it. It's the right approach, for sure tho. I just
wanted to make sure that someone more familiar with that bit of code
weighed in on it.
On Mar 25, 2016 6:24 AM, "Fedor Indutny" <notifications@github.com> wrote:

> In lib/_http_incoming.js
> <https://github.com/nodejs/node/pull/5844#discussion_r57443293>:
>
> > @@ -121,7 +121,7 @@ IncomingMessage.prototype._addHeaderLines = function(headers, n) {
> >  // and drop the second. Extended header fields (those beginning with 'x-') are
> >  // always joined.
> >  IncomingMessage.prototype._addHeaderLine = function(field, value, dest) {
> > -  field = field.toLowerCase();
> > +  field = field.toLowerCase().trim();
>
> @jasnell <https://github.com/jasnell> why is it hard to do it in
> http-parser? It seems to me that it is just a matter of adding two extra
> states.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/5844/files/2f45d2eda4f16435bf844ebf4f28e84bf159e2cb#r57443293>
>